packages/artifact/src/internal-artifact-client.ts

```diff
@@ -110,23 +106,27 @@ export class DefaultArtifactClient implements ArtifactClient {
       core.debug(`Upload Resource URL: ${response.fileContainerResourceUrl}`)
 
       // Upload each of the files that were found concurrently
-      const uploadResult = await uploadArtifactToFileContainer(
+      const uploadResult = await uploadHttpClient.uploadArtifactToFileContainer(
         response.fileContainerResourceUrl,
         uploadSpecification,
         options
       )
 
-      //Update the size of the artifact to indicate we are done uploading
-      await patchArtifactSize(uploadResult.size, name)
+      // Update the size of the artifact to indicate we are done uploading
+      // The uncompressed size is used for display when downloading a zip of the artifact from the UI
+      await uploadHttpClient.patchArtifactSize(
+        uploadResult.uncompressedSize,
```

Comment:
This has been a hotly debated topic - if we upload compressed and charge for _uncompressed_, it's odd.  To reduce questions, I would call this `totalSize` instead of calling out the compressiveness.